lib/reek/smells/nested_iterators.rb

```diff
@@ -37,40 +36,45 @@ def self.default_config
       end
 
       #
-      # Attempts to find the deepest nested iterator and warns if it's depth
+      # Attempts to find the deepest nested iterator and warns if its depth
       # is bigger than our allowed maximum.
       #
       # @return [Array<SmellWarning>]
       #
-      # :reek:TooManyStatements: { max_statements: 6 }
       def inspect(ctx)
-        configure_ignore_iterators(ctx)
-        deepest_iterator = find_deepest_iterator ctx
-        return [] unless deepest_iterator
-        depth = deepest_iterator.depth
-        return [] unless depth > max_nesting(ctx)
+        configure_ignore_iterators ctx
+        violations = find_violations ctx
 
-        [smell_warning(
-          context: ctx,
-          lines: [deepest_iterator.exp.line],
-          message: "contains iterators nested #{depth} deep",
-          parameters: { name: ctx.full_name, count: depth })]
+        violations.group_by(&:depth).map do |depth, group|
+          lines = group.map(&:line)
+          smell_warning(
+            context: ctx,
+            lines: lines,
+            message: "contains iterators nested #{depth} deep",
+            parameters: { name: ctx.full_name, count: depth })
+        end
       end
 
       private
 
       attr_accessor :ignore_iterators
 
+      # Finds deepest iterators that are nested more deeply than allowed.
+      #
+      # @return [Array<Iterator>]
+      #
+      def find_violations(ctx)
+        deepest_iterators = find_deepest_iterators ctx
```

Comment:
Well, it's the minimal set of independent deepest iterators (which may not be deep enough to count as violations), but that's a bit long. Perhaps I can call them `candidates` instead?